pydyad/pydyad/bindings.py

```diff
@@ -0,0 +1,207 @@
+import ctypes
+import os
+from pathlib import Path
+import warnings
+
+from numpy.ctypeslib import load_library
+
+
+DYAD_LIB_DIR = None
+
+
+class FluxHandle(ctypes.Structure):
+    pass
+
+
+class DyadDTLHandle(ctypes.Structure):
+    pass
+
+
+class DyadCtxWrapper(ctypes.Structure):
+    _fields_ = [
+        ("h", ctypes.POINTER(FluxHandle)),
+        ("dtl_handle", ctypes.POINTER(DyadDTLHandle)),
+        ("debug", ctypes.c_bool),
+        ("check", ctypes.c_bool),
+        ("reenter", ctypes.c_bool),
+        ("initialized", ctypes.c_bool),
+        ("shared_storage", ctypes.c_bool),
+        ("sync_started", ctypes.c_bool),
+        ("key_depth", ctypes.c_uint),
+        ("key_bins", ctypes.c_uint),
+        ("rank", ctypes.c_uint32),
+        ("kvs_namespace", ctypes.c_char_p),
+        ("prod_managed_path", ctypes.c_char_p),
+        ("cons_managed_path", ctypes.c_char_p),
+    ]
+
+
+class Dyad:
+
+    def __init__(self):
+        self.dyad_core_lib = None
+        self.ctx = None
+        self.dyad_init = None
+        self.dyad_init_env = None
+        self.dyad_produce = None
+        self.dyad_consume = None
+        self.dyad_finalize = None
+        dyad_core_libpath = None
+        self.cons_path = None
+        self.prod_path = None
+        if DYAD_LIB_DIR is None:
```

Comment:
Apologies in advance if I'm just asking bad questions - this is the first time I'm looking here (and I'm excited to try this out). Is there any reason the user is required to export `LD_LIBRARY_PATH`? Can we not do the same as the linker and look in the ld so config or default library paths? I read up on this sometime last year and was using this class to derive paths: https://github.com/vsoch/elfcall/blob/dc7383ecd6386cff9927bbf4b3b65335a45f97f4/elfcall/main/ld.py#L37

Reply:
TBH, this was just an arbitrary decision so that I'd have an initial way to locate `libdyad_core.so`. I can definitely replace this with a linker-style search.

Reply:
Awesome! Feel free to use the class in elfcall it’s published on pypi.